api/v1alpha1/profilecatalogsource_types.go

```diff
@@ -19,13 +19,30 @@ package v1alpha1
 import (
 	"strings"
 
+	"github.com/fluxcd/pkg/apis/meta"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
 // ProfileCatalogSourceSpec defines the desired state of ProfileCatalogSource
 type ProfileCatalogSourceSpec struct {
 	// Profiles is the list of profiles exposed by the catalog
+	// +optional
 	Profiles []ProfileCatalogEntry `json:"profiles,omitempty"`
+	// Repositories is the list of repositories to scan for profiles
+	// +optional
+	Repos []Repository `json:"repositories,omitempty"`
+}
+
+type Repository struct {
+	// URL is the URL of the repository
+	URL string `json:"url,omitempty"`
+	// The secret name containing the Git credentials.
+	// For HTTPS repositories the secret must contain username and password
+	// fields.
+	// For SSH repositories the secret must contain identity, identity.pub and
+	// known_hosts fields.
+	// +optional
+	SecretRef *meta.LocalObjectReference `json:"secretRef,omitempty"`
 }
 
 // ProfileDescription defines details about a given profile.
```

Comment:
err, I know it's not your concern, but I just noticed that this isn't the right description for this struct. :D Would you mind? Pretty please? 🙇

Reply:
is it not? I copied from flux https://github.com/fluxcd/source-controller/blob/2dc713258d3c4cae91923dc9689e6b468de1d54e/api/v1beta1/gitrepository_types.go#L42-L48

Reply:
? This is comment is for the struct below this comment. :D

Reply:
oooh lol